### PR TITLE
[SDL2] Move sources and documentation in subfolders

### PR DESCRIPTION
**Pull request by madebr**
Addresses the SDL2 branch for #369

Untested changes include:
- Android.mk
- VC and VC-UWP
- xcode project
- Makefile.os2

### CODE REVIEW

**Comment by madebr** [comment]
I left `SDL_image.h` in the root folder.
I don't think it's worth to change this at the end of SDL2_image's life cycle.

File: Xcode/SDL_image.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -652,6 +652,8 @@
 				);
 				GCC_SYMBOLS_PRIVATE_EXTERN = YES;
 				HEADER_SEARCH_PATHS = (
+					"\"$(SRCROOT)/include\"",
+					"\"$(SRCROOT)/src\"",
```
Comment (madebr) [inline Xcode/SDL_image.xcodeproj/project.pbxproj:656]:
```suggestion
					"\"$(SRCROOT)/../include\"",
					"\"$(SRCROOT)/../src\"",
```

File: Xcode/SDL_image.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -741,6 +743,8 @@
...
```
Comment (madebr) [inline Xcode/SDL_image.xcodeproj/project.pbxproj:747]:
```suggestion
					"\"$(SRCROOT)/../include\"",
					"\"$(SRCROOT)/../src\"",
```

File: aclocal.m4 [hunk 3]
```diff
@@ -20,6 +20,106 @@ You have another version of autoconf.  It may work, but is not guaranteed to.
...
+m4_ifndef([AC_LIBTOOL_CONFIG],		[AC_DEFUN([AC_LIBTOOL_CONFIG])])
+m4_ifndef([_LT_AC_FILE_LTDLL_C],	[AC_DEFUN([_LT_AC_FILE_LTDLL_C])])
+m4_ifndef([_LT_REQUIRED_DARWIN_CHECKS],	[AC_DEFUN([_LT_REQUIRED_DARWIN_CHECKS])])
+m4_ifndef([_LT_AC_PROG_CXXCPP],		[AC_DEFUN([_LT_AC_PROG_CXXCPP])])
+m4_ifndef([_LT_PREPARE_SED_QUOTE_VARS],	[AC_DEFUN([_LT_PREPARE_SED_QUOTE_VARS])])
+m4_ifndef([_LT_PROG_ECHO_BACKSLASH],	[AC_DEFUN([_LT_PROG_ECHO_BACKSLASH])])
+m4_ifndef([_LT_PROG_F77],		[AC_DEFUN([_LT_PROG_F77])])
+m4_ifndef([_LT_PROG_FC],		[AC_DEFUN([_LT_PROG_FC])])
+m4_ifndef([_LT_PROG_CXX],		[AC_DEFUN([_LT_PROG_CXX])])
+
```
Comment (madebr) [inline aclocal.m4:122]:
```suggestion
```

Comment (sezero) [reply aclocal.m4]: libtool somehow manages to do that crap: yes, it can be removed.

**Pull request merged**